privatenetworks_test.go

```diff
@@ -0,0 +1,199 @@
+		"ipv6aggregate": "fe00:123::/48"
+		} } }`}
+	n := PrivateNetworkService{client: c}
+
+	net, _ := n.Details(context.Background(), "pn-123ab")
+
+	assert.Equal(t, "POST", c.lastMethod, "method used is correct")
+	assert.Equal(t, "privatenetwork/details", c.lastPath, "path used is correct")
+	assert.Equal(t, "mynetwork", net.Name, "Name is correct")
+	assert.Equal(t, "pn-123ab", net.ID, "ID is correct")
+	assert.Equal(t, "fe00:123::/48", net.IPv6Aggregate, "IPv6Aggregate is correct")
+}
+
+func TestPrivateNetworksList(t *testing.T) {
+	c := &mockClient{body: `{ "response": { "privatenetworks": [{
+		"id": "pn-123ab",
+		"name": "mynetwork",
+		"ipv6aggregate": "fe00:123::/48"
+		}, {
+		"id": "pn-456cd",
+		"name": "othernet",
+		"ipv6aggregate": "fe00:456::/48" } ] } }`}
+
+	n := PrivateNetworkService{client: c}
+
+	net, _ := n.List(context.Background())
+
+	assert.Equal(t, "POST", c.lastMethod, "method used is correct")
+	assert.Equal(t, "privatenetwork/list", c.lastPath, "path used is correct")
+	assert.Equal(t, "mynetwork", (*net)[0].Name, "Name is correct")
+	assert.Equal(t, "pn-123ab", (*net)[0].ID, "ID is correct")
+	assert.Equal(t, "fe00:123::/48", (*net)[0].IPv6Aggregate, "IPv6Aggregate is correct")
+	assert.Equal(t, "othernet", (*net)[1].Name, "Name is correct")
+	assert.Equal(t, "pn-456cd", (*net)[1].ID, "ID is correct")
+	assert.Equal(t, "fe00:456::/48", (*net)[1].IPv6Aggregate, "IPv6Aggregate is correct")
+}
+
+func TestPrivateNetworksEdit(t *testing.T) {
+	c := &mockClient{body: `{ "response": { "privatenetwork":
+		{ "name": "newnetwork-1", "id": "pn-123ab"}}}`}
+
+	n := PrivateNetworkService{client: c}
+
+	params := EditPrivateNetworkParams{
+		Name: "newnetwork-1",
+		ID:   "pn-123ab",
+	}
+
+	net, _ := n.Edit(context.Background(), params)
+
+	assert.Equal(t, "POST", c.lastMethod, "method is used correct")
+	assert.Equal(t, "privatenetwork/edit", c.lastPath, "path used is correct")
+	assert.Equal(t, "pn-123ab", net.ID, "ID is correct")
+	assert.Equal(t, "newnetwork-1", net.Name, "Name is correct")
+}
+
+func TestPrivateNetworksEstimatedcost(t *testing.T) {
+	c := &mockClient{body: `{ "response": { "billing":
+		{ "currency": "SEK", "price": 50.0,
+		"discount": 12.5, "total": 37.5}}}`}
+
+	n := PrivateNetworkService{client: c}
+
+	netid := "pn-123ab"
+
+	net, _ := n.EstimatedCost(context.Background(), netid)
+
+	assert.Equal(t, "POST", c.lastMethod, "method is used correct")
+	assert.Equal(t, "privatenetwork/estimatedcost", c.lastPath, "path used is correct")
+	assert.Equal(t, 50.0, net.Price, "Price is correct")
+	assert.Equal(t, 12.5, net.Discount, "Discount is correct")
+	assert.Equal(t, 37.5, net.Total, "Total price is correct")
+}
+
+func TestPrivateNetworkSegmentsCreate(t *testing.T) {
+	c := &mockClient{body: `{ "response": { "privatenetworksegment":
+		{ "id": "266979ab-1e05-4fbc-b9e0-577f31c0d2e9",
+		"name": "mysegment", "ipv6subnet": "fe00:123:0:/64",
+		"ipv4subnet": "10.0.0.0/24", "datacenter": "dc-fbg1",
```

Comment:
```suggestion
		"ipv4subnet": "192.168.0.0/24", "datacenter": "dc-fbg1",
```

Use any of the TEST-NET ranges - 192.168.0.0/16, 198.51.100.0/24, 203.0.113.0/24